core-web/libs/sdk/experiments/src/lib/dot-experiments.ts

```diff
@@ -239,7 +266,84 @@ export class DotExperiments {
      * @return {void}
      */
     public trackPageView(): void {
+        if (!this.shouldTrackPageView()) {
+            this.logger.log(`No trackPageView triggered.`);
+
+            return;
+        }
+
         this.track(PAGE_VIEW_EVENT_NAME);
+
+        this.prevLocation = this.currentLocation.href;
+    }
+
+    /**
+     * This method is used to retrieve the variant associated with a given URL.
+     *
+     * It checks if the URL is part of an experiment by verifying it against an experiment's regex. If the URL matches the regex of an experiment,
+     * it returns the variant attached to that experiment; otherwise, it returns null.
+     *
+     * @param {string | null} path - The URL to check for a variant. This should be the path of the URL.
+     *
+     * @returns {Variant | null} The variant associated with the URL if it exists, null otherwise.
+     */
+    public getVariantFromHref(path: string | null): Variant | null {
```

Comment:
We should consider using the "guard clause" pattern.

Catch errors or empty returns should be handled first, with the main logic at the end of the function

```ts
public getVariantFromHref(path: string | null): Variant | null {
    if (!(this.experimentsAssigned && path)) {
        return null;
    }

    const experiment = this.experimentsAssigned.find((experiment) => {
        const url = getFullUrl(this.currentLocation, path) ?? '';

        return url ? verifyRegex(experiment.regexs.isExperimentPage, url) : false;
    });

    return experiment ? experiment.variant : null;

}
```